src/flows/components/header/index.tsx

```diff
@@ -133,6 +141,53 @@ const FlowHeader: FC = () => {
       .catch(err => console.error('failed to get notebook share', err))
   }, [flow.id])
 
+  const handlePublish = useCallback(() => {
+    if (isFlagEnabled('flowPublishLifecycle')) {
+      // TODO(ariel): move this to openAPI
+      setPublishLoading(RemoteDataState.Loading)
+      fetch(`/api/v2private/notebooks/${flow.id}/version`, {
+        method: 'POST',
+      })
+        .then(resp => resp.json())
+        .then(resp => {
+          if (!resp.length) {
+            return
+          }
+        })
+        .catch(error => {
+          console.error({error})
+          setPublishLoading(RemoteDataState.Error)
+        })
+      setPublishLoading(RemoteDataState.Done)
```

Comment:
i think you wanna move this into the last then or add an await to the fetch

Reply:
totally. Working on the openapi stuff now